dipy/reconst/dti.py

```diff
@@ -777,17 +782,35 @@ def fit(self, data, mask=None):
             min_signal = self.min_signal
 
         data_in_mask = np.maximum(data_in_mask, min_signal)
-        params_in_mask = self.fit_method(self.design_matrix, data_in_mask,
-                                         *self.args, **self.kwargs)
+
+        if self.return_S0_hat:
+            params_in_mask, model_S0 = self.fit_method(
+                self.design_matrix,
+                data_in_mask,
+                return_S0_hat=self.return_S0_hat,
+                *self.args,
+                **self.kwargs)
+        else:
+            params_in_mask = self.fit_method(self.design_matrix,
+                                             data_in_mask,
+                                             *self.args, **self.kwargs)
 
         if mask is None:
             out_shape = data.shape[:-1] + (-1, )
             dti_params = params_in_mask.reshape(out_shape)
+            if self.return_S0_hat:
```

Comment:
Yes - I would make `S0_params` an optional input to `fit` with a default of None. Does that make sense? But you're right that I don't see a way to simplify more.